pkg/target/locker.go

```diff
@@ -34,14 +34,14 @@ type Locker interface {
 	// Locks are reentrant, locking existing locks (with the same owner)
 	// extends the deadline.
 	Lock(types.JobID, []*Target) error
-	// TryLock attempts to lock all given targets.
+	// TryLock attempts to lock up to limit of the given targets.
 	// The job ID is the owner of the lock.
-	// This function attempts to lock as many of the given targets as possible,
+	// This function attempts to lock up to limit of the given targets,
 	// and returns a list of target IDs that it was able to lock.
 	// This function does not return an error if it was not able to lock any targets.
 	// Locks are reentrant, locking existing locks (with the same owner)
 	// extends the deadline.
-	TryLock(types.JobID, []*Target) ([]string, error)
+	TryLock(jobID types.JobID, targets []*Target, limit int) ([]string, error)
```

Comment:
could this be `uint`? In which case you can also get rid of the `<= 0` check below that every plugin will have to implement (the case when it's 0 is still up though)

Reply:
Switched to `uint` because people seem to like it better.

I still disagree :)
The API does indeed look nicer and it makes the purpose of the parameter more clear, but now it is more cumbersome for _both_ the user and the implementer as one needs to cast to `uint` all the time, since builtins like `len()` return `int`.

I don't think any type-safety argument can be made here, as it will happily cast negative values to huge integers without notice at runtime, my view is one might as well accept `int` and implement some sane behaviour (or clear exception) for negative values.

Note this change to `uint` creates a subtile change in behavior. Before, plugins would not lock any targets on negative values. Now an accidentally casted negative value is a huge integer, so the plugin will lock many/all targets.